Client/game_sa/CBuildingsPoolSA.cpp

```diff
@@ -0,0 +1,178 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        game_sa/CBuildingsPoolSA.cpp
+ *  PURPOSE:     Buildings pool class
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "CBuildingsPoolSA.h"
+
+#include "CFileLoaderSA.h"
+#include <game/CWorld.h>
+#include "CGameSA.h"
+#include "CPtrNodeSingleListSA.h"
+
+extern CGameSA* pGame;
+
+class CClientEntity;
+
+CBuildingsPoolSA::CBuildingsPoolSA() : m_pOriginalBuildingsBackup(nullptr)
+{
+    m_ppBuildingPoolInterface = (CPoolSAInterface<CBuildingSAInterface>**)0xB74498;
+}
+
+inline bool CBuildingsPoolSA::AddBuildingToPool(CClientBuilding* pClientBuilding, CBuildingSA* pBuilding)
+{
+    // Grab the new object interface
+    CBuildingSAInterface* pInterface = pBuilding->GetBuildingInterface();
+
+    if (!pInterface)
+        return false;
+
+    uint32_t dwElementIndexInPool = (*m_ppBuildingPoolInterface)->GetObjectIndexSafe(pInterface);
+    if (dwElementIndexInPool >= -1)
```

Comment:
`dwElementIndexInPool` cannot be bigger than -1(0xffffffff). It's better to leave it `dwElementIndexInPool == -1`. Though, I personally prefer `dwElementIndexInPool == UINT_MAX` or `dwElementIndexInPool == std::numeric_limits<uint32_t>().max()`

Reply:
The same at line 93.